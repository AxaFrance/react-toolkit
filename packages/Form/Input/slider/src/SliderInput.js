import React from 'react';
import PropTypes from 'prop-types';
import {
  Field,
  FieldConstants as Constants,
  HelpMessage,
} from '@axa-fr/react-toolkit-form-core';
import { InputManager } from '@axa-fr/react-toolkit-core';

import Slider from './Slider';

const propTypes = {
  ...Constants.propTypes,
  options: PropTypes.array.isRequired,
};

const defaultProps = {
  ...Constants.defaultProps,
};

const SliderInput = props => {
  const {
    id,
    children,
    helpMessage,
    message,
    classNameContainerLabel,
    classNameContainerInput,
    label,
    messageType,
    isVisible,
    forceDisplayMessage,
    className,
    classModifier,
    name,
    value,
    onChange,
    readOnly,
    disabled,
    placeholder,
    ...otherProps
  } = props;
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      label={label}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <Slider
        name={name}
        id={inputId}
        value={value}
        onChange={onChange}
        readOnly={readOnly}
        disabled={disabled}
        placeholder={placeholder}
        classModifier={classModifier}
        {...otherProps}
      />
      {children}
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

SliderInput.propTypes = propTypes;
SliderInput.defaultProps = defaultProps;

export default SliderInput;
