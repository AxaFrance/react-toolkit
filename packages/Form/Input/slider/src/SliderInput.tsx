import React, { ComponentProps } from 'react';
import {
  Field,
  HelpMessage,
  InputManager,
} from '@axa-fr/react-toolkit-form-core';

import Slider from './Slider';

type Props = ComponentProps<typeof Field> &
  ComponentProps<typeof Slider> & {
    helpMessage?: string;
  };

const SliderInput = ({
  id,
  children,
  helpMessage,
  message,
  classNameContainerLabel,
  classNameContainerInput,
  label,
  messageType,
  isVisible,
  forceDisplayMessage,
  className,
  classModifier,
  ...sliderProps
}: Props) => {
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      id={inputId}
      label={label}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <Slider {...sliderProps} id={id} />
      {children}
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

export default SliderInput;
