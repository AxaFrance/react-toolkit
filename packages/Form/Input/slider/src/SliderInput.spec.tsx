import React from 'react';
import { shallow } from 'enzyme';
import SliderInput from './SliderInput';

const options = [
  {
    value: 15,
    label: '15',
  },
  { value: 1, label: 1 },
];

describe('<SliderInput>', () => {
  it('renders SliderInput correctly', () => {
    const wrapper = shallow(
      <SliderInput
        options={options}
        label="File *"
        id="id"
        name="file"
        onChange={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
