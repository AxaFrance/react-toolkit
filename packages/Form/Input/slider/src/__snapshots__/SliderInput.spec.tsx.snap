// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderInput> renders SliderInput correctly 1`] = `
<Field
  className="row af-form__group"
  classNameContainerInput="col-md-10"
  classNameContainerLabel="col-md-2"
  forceDisplayMessage={false}
  id="id"
  isVisible={true}
  label="File *"
  messageType="error"
>
  <Component
    id="id"
    name="file"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "15",
          "value": 15,
        },
        Object {
          "label": 1,
          "value": 1,
        },
      ]
    }
  />
  <HelpMessage
    isVisible={true}
  />
</Field>
`;
